apps/fishing-map/features/datasets/datasets.slice.ts

```diff
@@ -73,43 +73,56 @@ export const fetchDatasetByIdThunk = createAsyncThunk<
   }
 })
 
+const fetchDatasetsFromApi = async (
+  ids: string[] = [],
+  existingIds: string[] = [],
+  signal: AbortSignal,
+  maxDepth: number = 5
```

Comment:
I'd set up this as `maxDepth=2` to ensure we won't go deeper than two levels. 
This way we only go to the related of a related instead of fetching more datasets than probably needed

Reply:
Activity => vessel info => vessel track is

Reply:
Sounds good, let's keep 5 although I don't think we won't ever need it as the deeper relationship we have is:
`activity => info => track` and I don't think that will change we will be safe in the future